lib/upgradedb/mysql.2016102500.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * LMS version 1.11-git
+ *
+ *  (C) Copyright 2001-2016 LMS Developers
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License Version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+ *  USA.
+ *
+ */
+
+function splitNameToFirstAndLastName($users_before) {
+    foreach ($users_before as $id => $user) {
+        $parts = explode(' ', $user['name']);
+        $lastname = array_pop($parts);
+        $firstname = implode(' ', $parts);
+        $users[$id]['id'] = $user['id'];
+        $users[$id]['firstname'] = $firstname;
+        $users[$id]['lastname'] = $lastname;
+    }
+    return $users;
+}
+
+$this->BeginTrans();
+
+$this->Execute("ALTER TABLE users ADD COLUMN firstname varchar(64) NOT NULL DEFAULT ''");
+$this->Execute("ALTER TABLE users ADD COLUMN lastname varchar(64) NOT NULL DEFAULT ''");
+
+$users_before = $this->GetAll('SELECT id, name FROM users');
+$users = splitNameToFirstAndLastName($users_before);
+foreach ($users as $user) {
+    $this->Execute("UPDATE users SET firstname=?, lastname=? WHERE id = ?", array($user['firstname'], $user['lastname'], $user['id']));
+}
+$this->Execute("ALTER TABLE users DROP COLUMN name");
+
+$this->Execute("DROP VIEW vusers");
+$this->Execute("CREATE VIEW vusers AS SELECT *, CONCAT(firstname, ' ', lastname) AS name, CONCAT(lastname, ' ', firstname) AS rname FROM users");
```

Comment:
Chyba dopadło mnie jesienne przesilenie
```php
public function GroupConcat($field, $separator = ',');
```
nijak mi to nie pasuje...
https://github.com/lmsgit/lms/blob/1dddbab05fcbedcd4ec02ba8e1d1aa3bc384d202/lib/LMSDBInterface.php#L89